demo_callbacks.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2024 D-Wave
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import NamedTuple, Union
+
+import dash
+from dash import MATCH, ctx
+from dash.dependencies import Input, Output, State
+from dash.exceptions import PreventUpdate
+import plotly.graph_objs as go
+from plotly.subplots import make_subplots
+
+from data import DataSet
+from demo_interface import generate_problem_details_table_rows
+from src.demo_enums import SolverType
+from src.utils import draw_bar_chart, draw_accuracy_bars
+
+
+@dash.callback(
+    Output({"type": "to-collapse-class", "index": MATCH}, "className"),
+    inputs=[
+        Input({"type": "collapse-trigger", "index": MATCH}, "n_clicks"),
+        State({"type": "to-collapse-class", "index": MATCH}, "className"),
+    ],
+    prevent_initial_call=True,
+)
+def toggle_left_column(collapse_trigger: int, to_collapse_class: str) -> str:
+    """Toggles a 'collapsed' class that hides and shows some aspect of the UI.
+
+    Args:
+        collapse_trigger (int): The (total) number of times a collapse button has been clicked.
+        to_collapse_class (str): Current class name of the thing to collapse, 'collapsed' if not
+            visible, empty string if visible.
+
+    Returns:
+        str: The new class name of the thing to collapse.
+    """
+
+    classes = to_collapse_class.split(" ") if to_collapse_class else []
+    if "collapsed" in classes:
+        classes.remove("collapsed")
+        return " ".join(classes)
+    return to_collapse_class + " collapsed" if to_collapse_class else "collapsed"
+
+
+@dash.callback(
+        Output("features", "max"),
+        Output("features", "value"),
+        Output("features", "marks"),
+        inputs=[
+            Input("dataset", "value"),
+        ],
+)
+def create_features_input(data_set: str):
```

Comment:
missing return types here

```suggestion
def create_features_input(data_set: str) -> tuple[int, int, dict]:
```